Remotion/Validation/UnitTests/Implementation/CompoundValidationMessageFactoryTest.cs

```diff
@@ -14,23 +14,160 @@
 // You should have received a copy of the GNU Lesser General Public License
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
+using Moq;
 using NUnit.Framework;
+using Remotion.Reflection;
+using Remotion.Validation.Implementation;
+using Remotion.Validation.UnitTests.Implementation.TestDomain;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Implementation
 {
   [TestFixture]
   public class CompoundValidationMessageFactoryTest
   {
     [Test]
-    [Ignore("RM-5906")]
-    public void CreateValidationMessageForPropertyValidator ()
+    public void CreateValidationMessageForPropertyValidator_ContainsMessageFromSavedFactory ()
     {
+      var validatorStub = new Mock<IPropertyValidator>();
+      var propertyStub = new Mock<IPropertyInformation>();
+
+      var expectedMessage = new FakeValidationMessage("Oh no, a validation error!");
+
+      var factoryStub = new Mock<IValidationMessageFactory>();
+      factoryStub
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object))
+          .Returns(expectedMessage);
+
+      var factory = new CompoundValidationMessageFactory(new[] { factoryStub.Object });
+
+      var result = factory.CreateValidationMessageForPropertyValidator(validatorStub.Object, propertyStub.Object);
+
+      Assert.That(result, Is.EqualTo(expectedMessage));
+    }
+
+    [Test]
+    public void CreateValidationMessageForPropertyValidator_WithSeveralFactories_ContainsMessageFromFirstFactory ()
```

Comment:
```suggestion
    public void CreateValidationMessageForPropertyValidator_WithMultipleFactories_ReturnsMessageFromFirstFactory ()
```

Multiple vs Several: stylistic thing, we don't use "several", I think.